addons/event/models/event_registration.py

```diff
@@ -238,6 +238,12 @@ def _phone_format(self, number, country):
     # ACTIONS / BUSINESS
     # ------------------------------------------------------------
 
+    def action_previous_state(self):
+        if self.state == 'open':
+            self.write({'state': 'draft'})
+        elif self.state == 'done':
+            self.write({'state': 'open'})
```

Comment:
Would use action method if possible, like
```suggestion
    def action_set_previous_state(self):
        self.filtered(lambda reg: reg.state == 'open').action_set_draft()
        self.filtered(lambda reg: reg.state == 'done').action_confirm()
```